### PR TITLE
[SDK] Test coverage

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR focuses on enhancing the `padBytes` function in `to-bytes.ts` for better testing and functionality, while also adding comprehensive test cases for various encoding utilities.

### Detailed summary
- Added JSDoc comment to `padBytes` in `to-bytes.ts`.
- Exported `padBytes` for testing.
- Introduced multiple test cases for `byteSize`, `assertSize`, `charCodeToBase16`, `isHex`, and `toBytes` functions.
- Enhanced testing for `boolToBytes`, `hexToBytes`, `numberToBytes`, and `stringToBytes` functions.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#5316** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/utils/encoding/from-bytes.test.ts
Comment (jnsdls) [inline packages/thirdweb/src/utils/encoding/from-bytes.test.ts:1]: remove empty file

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: packages/thirdweb/src/utils/encoding/helpers/is-hext.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    expect(isHex("", { strict: false })).toBe(false);
+  });
+
+  it("should return true for valid empty hex prefix in non-strict mode", () => {
+    expect(isHex("0x", { strict: false })).toBe(true);
+  });
+
+  it("should return false for valid empty hex prefix in strict mode", () => {
+    expect(isHex("0x", { strict: true })).toBe(true); // Since it's valid according to regex
+  });
```
[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/thirdweb/src/utils/encoding/helpers/is-hext.test.ts:48]: checking

[the remark this answers is not in the document]
Comment (kien-ngo) [reply packages/thirdweb/src/utils/encoding/helpers/is-hext.test.ts]: fixed

**Review by jnsdls — APPROVED** [review]

**Pull request merged**